### ISSUE TITLE
Is it possible to get the minimal diameter of all threads?

### ISSUE THREAD

**Issue by tangruize** [tlaplus/tlaplus#738]
My colleague has encountered this situation:
* using 1 thread and running for several hours, the diameter is 13 and the states number is huge.
* using 10 threads and running for several minutes, the diameter is 16 and the states number is small.

After referring to [Specifying Systems](https://lamport.azurewebsites.net/tla/book-21-07-04.pdf), we find a footnote in Chapter 14.5.2 (page 254):

![Screenshot_20220612_170826](https://user-images.githubusercontent.com/19245736/173226477-07132986-403d-402f-9b85-5effff89ef95.png)

It seems that there is no guarantee all state are traversed under diameter depth. Is it possible to get the minimal diameter of all threads?

**Comment by lemmy** [comment]
The `TLCGet("level")` operator equals the diameter of the current state.

**Comment by tangruize** [comment]
I checked the [current-tool.pdf](https://lamport.azurewebsites.net/tla/current-tools.pdf) and found the definition of `level`:
> <b>"level"</b> The length of the path in the state graph from an initial state to the current state.

It seems that even if I get all levels of all paths, I still cannot find the minimal depth that all states under this depth are checked. Is it right?

**Comment by lemmy** [comment]
There should be a strong correlation between the minimal level of all levels reported by all workers and the actual minimal depth.  However, you are correct that there might still be (unexplored) states with a smaller diameter in the state queue.  The behavior you are seeing is unusual; can you share the spec & model?

**Comment by tangruize** [comment]
I finally find that the thread number setting is incorrect. There are 4 cores in my colleague's computer but workers number is set to 10.

Here is the process I tried to find the reason... It's strange that I failed to reproduce the model checking result in my computer using the exactly same (including seed) model parameters. (the diameter is just 13). I suspect different running environment, such as TLC/Java version, OS and threads number, may cause this result. After checking the log, I found that it uses 10 out of 4 threads. My computer has 20 cores. after setting to 40 threads, the diameter quickly reaches 15.
